pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -505,6 +506,10 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 
 	pod := c.pods.getPodByIP(proxyIP)
 	if pod != nil {
+		// for k8s multi cluster, in case there are pods of the same ip across clusters
```

Comment:
This seems super risky - what happens if this code is hit in single-cluster mode ? Any way to test this ?
Even for master, if we want to do this we should have some test - this area has very little coverage ( in particular on integration side ) and is very dangerous to make changes.

And in general: in a stable branch any change that may break existing users MUST have a setting to allow 
quick disabling. 

/hold

Reply:
>This seems super risky - what happens if this code is hit in single-cluster mode ? Any way to test this ?

There are already many tests for single cluster.